File: AngularApp/src/app/models/Cidade.ts
import { Cliente } from "./Cliente";
import { Empresa } from "./Empresa";
import { EnumEstado } from "./EnumEstado";
import { Fornecedor } from "./Fornecedor";

export interface Cidade{
    id?: number;
    descricao: string;
    estado: EnumEstado;
    estadoDescricao: string,
    clientesEndereco: Cliente;
    empresa: Empresa;
    fornecedor: Fornecedor;
}

File: ProjetoAspNetCoreApp/ClienteProduto.Domain/Entities/Cidade.cs
namespace ClienteProjeto.Domain.Entities;

public class Cidade : Entity
{
    public Cidade(int id, string descricao)
    {
        Id = id;
        Descricao = descricao;
    }

    public string Descricao { get; set; }
    public EnumEstado Estado { get; set;}
    public ICollection<Cliente> Clientes { get; set; }
    public ICollection<Empresa> Empresas { get; set; }
    public ICollection<Fornecedor> Fornecedores { get; set; }
}
